R-Interface/jasprcpp.cpp

```diff
@@ -157,38 +162,21 @@ void STDCALL jaspRCPP_init(const char* buildYear, const char* version, RBridgeCa
 	rInside[".baseCitation"]		= CSTRING_TO_R(baseCitation);
 	rInside[".resultsFont"]			= resultsFont;
 
-	jaspResults::setSendFunc(sendToDesktopFunction);
-	jaspResults::setPollMessagesFunc(pollMessagesFunction);
-	jaspResults::setBaseCitation(baseCitation);
-	jaspResults::setInsideJASP();
 
-	jaspRCPP_logString("Initializing jaspBase.\n");
-	jaspRCPP_parseEvalQNT("library(methods)");
-	jaspRCPP_parseEvalQNT("library(jaspBase)");
+	rInside[".sendToDesktopFunction"]	= Rcpp::XPtr<sendFuncDef>(&sendToDesktopFunction);
+	rInside[".pollMessagesFunction"]	= Rcpp::XPtr<pollMessagesFuncDef>(&pollMessagesFunction);
+	rInside[".baseCitation"]			= baseCitation;
+	jaspRCPP_parseEvalQNT("jaspBase:::setSendFunc(.sendToDesktopFunction)");
+	jaspRCPP_parseEvalQNT("jaspBase:::setPollMessagesFunc(.pollMessagesFunction)");
+	jaspRCPP_parseEvalQNT("jaspBase:::setBaseCitation(.baseCitation)");
+	jaspRCPP_parseEvalQNT("jaspBase:::setInsideJasp()");
 
-	jaspRCPP_logString("Initializing jaspResultsModule.\n");
-
-	rInside["jaspResultsModule"]			= givejaspResultsModule();
-
-	//Adding some functions in R to the RefClass (generator) in the module
-	//To do: move this entirely to zzzWrapper if this wasn't done yet.
-	jaspRCPP_parseEvalQNT("jaspResultsModule$jaspTable$methods(addColumnInfo = function(name=NULL, title=NULL, overtitle=NULL, type=NULL, format=NULL, combine=NULL) { addColumnInfoHelper(name, title, type, format, combine, overtitle) })");
-	jaspRCPP_parseEvalQNT("jaspResultsModule$jaspTable$methods(addFootnote =   function(message='', symbol=NULL, col_names=NULL, row_names=NULL) { addFootnoteHelper(message, symbol, col_names, row_names) })");
-			
-	jaspRCPP_logString("Loading auxillary R-files.\n");
-
-#if defined _WIN32 || defined __MINGW32__
-	jaspRCPP_parseEvalQNT("source(file='Modules/writeImage.R')");
-	jaspRCPP_parseEvalQNT("source(file='Modules/zzzWrappers.R')");
-	jaspRCPP_parseEvalQNT("source(file='Modules/workarounds.R')");
-#else
-	jaspRCPP_parseEvalQNT("source(file='../Modules/writeImage.R')");
-	jaspRCPP_parseEvalQNT("source(file='../Modules/zzzWrappers.R')");
-	jaspRCPP_parseEvalQNT("source(file='../Modules/workarounds.R')");
-#endif
+//	jaspRCPP_logString("Initializing jaspBase.\n");
+//	jaspRCPP_parseEvalQNT("library(jaspBase)");
 
-	jaspRCPP_logString("initEnvironment().\n");
-	jaspRCPP_parseEvalQNT("initEnvironment()");
+//	if we have a separate engine for each module then we should move these kind of hacks to the .onAttach() of each module (instead of loading BayesFactor when Descriptives is requested).
```

Comment:
>It would load the first one it finds in the libPaths. I think the main R library is the last one in the libPaths, so it's probably not loading that one.

But this code is run on engine initialization so it doesnt yet get the libpaths for a module. And thus loads the one from mainlibrary.

The rest of your comment sounds fine